### PR TITLE
fix(app security) Restrict allowed origins for requests

### PR DESCRIPTION
**Pull request by Vince-janv**
## Description
sonarcloud has objections to our very permissive CORS policy. This PR restricts it to any listed under the CORS_ORIGINS environment variable. If this is not set any origin is allowed.

### Changed
- The Flask app now only accepts requests from the given origins

### How to prepare for test

- [x] Deploy to stage

### How to test

- [x] Make sure that no CORS error is thrown for status-db or the orderportal.
- [x] Make sure that there is a cors error when sending a request from another domain, for instance test-cors.org

## Review

- [x] Tests executed by SA
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [x] Update the env file on cg-vm1
- [ ] Update the env file on cg-prod-services.scilifelab.se
- [ ] Document in Atlas
- [ ] Deploy to stage and prod

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,6 +12,8 @@ ENV CG_SQL_DATABASE_URI="sqlite:///:memory:"
 ENV CG_ENABLE_ADMIN="1"
 ENV CG_SECRET_KEY="key"
 
+ENV ALLOWED_ORIGINS=["https://some_site.com"]
```
Comment (henrikstranneheim) [inline Dockerfile:15]: How does this work? Should it not be list of allowed origin and not a mock one? I know very little about cors

**Comment by henrikstranneheim** [comment]
@northwestwitch Do you have any feedback on this one having worked with cors previously?

**Comment by northwestwitch** [comment]
> Do you have any feedback on this one having worked with cors previously?

We haven't set any ALLOWED_ORIGINS variable that I remember of.. For Scout we have this solution that looks quite similar to yours:

https://github.com/Clinical-Genomics/scout/commit/fc4aeca99692aee7c25e1210528e4f81dbc2eb5d

What's the original complain from sonarcloud?

**Comment by northwestwitch** [comment]
> > Do you have any feedback on this one having worked with cors previously?
> 
> We haven't set any ALLOWED_ORIGINS variable that I remember of.. For Scout we have this solution that looks quite similar to yours:
> 
> [Clinical-Genomics/scout@fc4aeca](https://github.com/Clinical-Genomics/scout/commit/fc4aeca99692aee7c25e1210528e4f81dbc2eb5d)
> 
> What's the original complain from sonarcloud?

Ah OK I found it: https://sonarcloud.io/project/security_hotspots?id=Clinical-Genomics_cg&file=cg%2Fserver%2Fext.py&fileUuid=AXDtcWqMYLFyXCsBNayz

File: cg/server/ext.py [hunk 2]
```diff
@@ -27,7 +27,7 @@ def init_app(self, app):
         super(FlaskLims, self).__init__(config)
 
 
-cors = CORS(resources={r"/api/*": {"origins": "*"}}, supports_credentials=True)
+cors = CORS(supports_credentials=True)
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply cg/server/ext.py:30]: Ah OK!

**Review by henrikstranneheim — APPROVED** [review]
👍

**Comment by seallard** [comment]
Added the following allowed origins to the env file on cg-vm1:
- https://clinical.scilifelab.se
- https://clinical-stage.scilifelab.se
- https://clinical-genomics-stage.web.app
- localhost

File: Dockerfile (hunk 1 above)
Comment (seallard) [reply Dockerfile]: https://github.com/Clinical-Genomics/atlas/pull/1481

**Pull request merged**